lib/puppet/provider/zabbix_host/ruby.rb

```diff
@@ -224,4 +235,37 @@ def proxy=(string)
       proxy_hostid: zbx.proxies.get_id(host: string)
     )
   end
+
+  def tls_connect=(int)
+    @property_hash[:tls_connect] = int
+    zbx.hosts.create_or_update(
+      host: @resource[:hostname],
+      tls_connect: @property_hash[:tls_connect].nil? ? 1 : @property_hash[:tls_connect]
+    )
+  end
+
+  def tls_accept=(int)
+    @property_hash[:tls_accept] = int
+
+    zbx.hosts.create_or_update(
+      host: @resource[:hostname],
+      tls_accept: @property_hash[:tls_accept].nil? ? 1 : @property_hash[:tls_accept]
+    )
+  end
+
+  def tls_issuer=(string)
+    @property_hash[:tls_issuer] = string
+    zbx.hosts.create_or_update(
+      host: @resource[:hostname],
+      tls_issuer: @property_hash[:tls_issuer].nil? ? '' : @property_hash[:tls_issuer]
+    )
+  end
+
+  def tls_subject=(string)
+    @property_hash[:tls_subject] = string
+    zbx.hosts.create_or_update(
+      host: @resource[:hostname],
+      tls_subject: @property_hash[:tls_subject].nil? ? '' : @property_hash[:tls_subject]
+    )
+  end
```

Comment:
These could refactored to use provider flushing instead of making individual API calls, probably faster and less resources are wasted. 
A topic for another PR since more properties have to be refactored 😄

Reply:
Yeah, it sure sounds like a great idea, but I’m not that familiar with custom provider, not so sure if I could pull it off! :smile: